examples/animation/animated_curve.rs

```diff
@@ -0,0 +1,265 @@
+//! Custom curve animation sampling
+use bevy::{animation::ActiveAnimation, input::common_conditions::input_just_pressed, prelude::*};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<AnimationControl>()
+        .add_systems(Startup, setup)
+        .add_systems(
+            Update,
+            (
+                keyboard_control,
+                pickup_sprite.run_if(input_just_pressed(MouseButton::Left)),
+                drop_sprite.run_if(input_just_pressed(MouseButton::Right)),
+                update_text.run_if(resource_changed::<AnimationControl>),
+            ),
+        )
+        .add_systems(PostUpdate, myanimation_system)
+        .run();
+}
+
+///update text
+#[derive(Component)]
+pub struct AnimationControlTextMark;
+
+/// Provide numerical values for animation.(computer translation)
+#[derive(Resource)]
+pub struct AnimationControl {
+    speed: f32,
+    distance: f32,
+}
+
+impl Default for AnimationControl {
+    fn default() -> Self {
+        Self {
+            speed: 100.0,
+            distance: 1.0,
+        }
+    }
+}
+
+#[derive(Component)]
+struct MyAnimationCurve {
+    active: ActiveAnimation,
+    // actually,I don't want use dyn
```

Comment:
It is not recommended to begin sentences with the first person in the code as it's too subjective.